internal/runner/run_result.go

```diff
@@ -0,0 +1,149 @@
+package runner
+
+import (
+	"github.com/buildkite/test-engine-client/internal/plan"
+)
+
+type RunStatus string
+
+const (
+	RunStatusPassed RunStatus = "passed"
+	RunStatusFailed RunStatus = "failed"
+	RunStatusError  RunStatus = "error"
+)
+
+// RunResult is a struct to keep track the results of a test run.
+// It contains the logics to record test results, calculate the status of the run.
+type RunResult struct {
+	// tests is a map containing individual test results.
+	tests map[string]*TestResult
+	// mutedTestLookup is a map containing the test identifiers of muted tests.
+	// This list might contain tests that are not part of the current run (i.e. belong to a different node).
```

Comment:
Why?

Reply:
The test plan currently returns muted tests for all nodes in a single field, it doesn't distribute the muted tests to the specific node. We don't want to display a muted test that belongs to a test file assigned for another node.